### PR TITLE
[Needs Review] - Changes to App server Advanced configuration for V2

### PR DESCRIPTION
**Pull request by DivergentEuropeans**
Signed-off-by: Leanid Astrakou <lastrakou@rocketsoftware.com>

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../docs/contribute/contributing.md) to this repository.

- [ ] If the changes in this PR is part of the next future release, make this pull request against the **docs-staging** branch which will be published at the next release boundary. If the changes in this PR are part of the current release, use the default base branch, **master**. For more information about branches, see https://github.com/zowe/docs-site/tree/master#understanding-the-doc-branches. 
      
- [ ] If this PR relates to GitHub issues in `docs-site` or other repositories, please list in Description, prefixed with **close**, **fix** or **resolve** keywords.

### Description (including links to related git issues)

This PR is a WIP for App server Advanced configuration change for V2. For the things that weren't too complicated I changed myself. The remainder I have compiled a list of sections that need to reviewed & these sections should have tasks created for them later

> You must use SSL certificates to configure the Zowe Application Server to communicate with the SSL-enabled Mediation Layer. Those certificates were created during the Zowe installation process, and are located in the $RUNTIME_DIR/components/app-server/share/zlux-app-server/defaults/serverConfig directory.

Those certificates are still there, but they now get generated during the zwe step. What do we do about the ones still there? We should specify information about zwe utility specific, i.e. update config will update certificate info in zowe.yaml

> ### Accessing the Application Server

docs-site\docs\user-guide\configure-instance-directory.md

All of this needs replacing

> ## Configuration file

Section may need review

> ## Network configuration

Need to replace examples with zowe.yaml usages

> ### Network example

Section may need total redo

> ### Old defaults

This whole section needs to be replaced
Edit: I replaced it. Someone should check that I did it right.

> ### Configuring ZSS for HTTPS

This section needs to be reviewed

> #### Creating certificates and key ring for the ZSS server using RACF

This section I don't know yet what needs to be changed

> #### Defining the AT-TLS rule
> ### Configuring AT-TLS on Client System

This section I don't know yet what needs to be changed

> #### Configuring the Zowe App Server for HTTPS communication with ZSS

I tried to edit this section but it certainly needs review

> ### Installing additional ZSS instances

This section I don't know yet what needs to be changed, other than removing references to instance directory

> ### Defining the RACF ZOWE class
> ### "from" Creating authorization profiles ---> "to" Configuring basic authorization

This section I believe is mostly kept the same, but it needs to be rereviewed

> ### Session duration and expiration
> ### Configuration

Section needs to be reviewed, as zosmf-auth, zss-auth, and apiml-auth are outdated and sso-auth is now solely used (I believe?)

> #### ZLUX_NODE_LOG_DIR and ZSS_LOG_DIR environment variables
> ### Retaining logs

I am unsure if this still works and should be checked.

> ## Administering the servers and plugins using an API

As I understand, our admin API's are supposed to still work for V2. But I haven't tested these, I don't know if anyone has tested these yet, and this section needs to re-reviewed

:heart:Thank you!

### CODE REVIEW

File: docs/extend/extend-desktop/mvd-logutility.md [hunk 1]
```diff
@@ -45,7 +45,9 @@ NodeJS uses `global` as its global object, so the logger is attached to:
 
 #### Web
 
-Browsers use `window` as the global object, so the logger is attached to:
+(Angular App Instance Injectible). See **Logger** in [Zowe Desktop and window management](mvd-desktopandwindowmgt.md).  
+
+(others) Browsers use `window` as the global object, so the logger is attached to:
 `window.COM_RS_COMMON_LOGGER`
```
Comment (DivergentEuropeans) [inline docs/extend/extend-desktop/mvd-logutility.md:51]: Shouldn't this become global.COM_RS_COMMON_LOGGER ?

**Review by DivergentEuropeans — COMMENTED** [review]
Note: There are still some references to INSTANCE_DIR to the sections I added in notes

File: docs/user-guide/mvd-configuration.md [hunk 2]
```diff
@@ -17,23 +17,22 @@ For earlier releases, you must register the Application Server with the Mediatio
 
 ### Accessing the Application Server
 
-**Note:** Before Accessing the Application Server, first [create and configure the Zowe instance directory](configure-instance-directory.md).
```
Comment (DivergentEuropeans) [inline docs/user-guide/mvd-configuration.md:20]: Change not relevant here, we need to totally redo configure-instance-directory.md anyway

File: docs/user-guide/mvd-configuration.md [hunk 3]
```diff
@@ -43,7 +42,7 @@ Follow these optional steps to configure the default connection to open for the
     type: <"telnet" or "tls">
   }
 ```
-### Setting up the VT Terminal application plug-in
+### Setting up the VT Terminal application plugin
```
Comment (DivergentEuropeans) [inline docs/user-guide/mvd-configuration.md:45]: It seems like some parts of the doc refer to "plug-ins" and some "plugins" I am changing everything to plugins to match what normally people use and what I see common in the community. Lmk if this is bad

File: docs/user-guide/mvd-configuration.md [hunk 4]
```diff
@@ -175,36 +189,42 @@ For more information, see [Logging Utility](../extend/extend-desktop/mvd-logutil
...
 
 You must have the authority to alter security definitions related to certificate management, and you must be authorized to work with and update the Policy Agent.
 
-To configure HTTPS communication between ZSS and the Zowe App Server, you need a key ring which contains the ZSS server certificate and its Certificate Authority (CA) certificate. You can use an internal CA to create the ZSS server certificate, or you can buy the ZSS server certificate from a well-known commercial Certificate Authority. Next you define an AT-TLS rule which points to the key ring used by the ZSS server. Then you copy the CA certificate to the Zowe App Server key store and update the Zowe App Server configuration file.
+By default, HTTPS communication between ZSS and App Server is done via a server certificate, its Certificate Authority (CA) certificate, and keystore generated during the Zowe initialization step (`zwe init`).
```
Comment (DivergentEuropeans) [inline docs/user-guide/mvd-configuration.md:225]: Remove 225

227 should say "If you are going to use AT-TLS, you will need to set up TLS rule and TLS keyring. The next section will cover that information."

File: docs/user-guide/mvd-configuration.md [hunk 5]
```diff
@@ -124,92 +145,150 @@ When running, the App Server will access the server's settings and read or modif
...
-### Directories example
-```
-// All paths relative to zlux-app-server/lib
-// In real installations, these values will be configured during the install.
-  "productDir":"../defaults",
-  "siteDir":"/home/myuser/.zowe/workspace/app-server/site",
-  "instanceDir":"/home/myuser/.zowe/workspace/app-server",
-  "groupsDir":"/home/myuser/.zowe/workspace/app-server/groups",
-  "usersDir":"/home/myuser/.zowe/workspace/app-server/users",
+These directories dictate where the Configuration Dataservice will store content. For more information, see the [Configuration Dataservice documentation](..//extend-desktop/mvd-configdataservice.md)
```
Comment (DivergentEuropeans) [inline docs/user-guide/mvd-configuration.md:148]: may be typo here "..//"

File: docs/user-guide/mvd-configuration.md [hunk 6]
```diff
@@ -701,158 +673,20 @@ The following is an example configuration for `zss-auth`, as seen in a default i
...
-
-### Retaining logs
-By default, the last five logs are retained. To specify a different number of logs to retain, set *ZLUX_NODE_LOGS_TO_KEEP* (Zowe Application Server logs) or *ZSS_LOGS_TO_KEEP* (ZSS logs) to the number of logs that you want to keep. For example, if you set *ZLUX_NODE_LOGS_TO_KEEP* to 10, when the eleventh log is created, the first log is deleted.
 
 ## Administering the servers and plugins using an API
-You can use a REST API to retrieve and edit Zowe Application Server and ZSS server configuration values, and list, add, update, and delete plugins. If an administrator has configured Zowe to [use RBAC](https://docs.zowe.org/stable/user-guide/mvd-configuration.html#applying-role-based-access-control-to-dataservices), they must authorize you to access the endpoints.
+You can use a REST API to retrieve and edit Zowe App Server and ZSS server configuration values, and list, add, update, and delete plugins. If an administrator has configured Zowe to [use RBAC](https://docs.zowe.org/stable/user-guide/mvd-configuration.html#applying-role-based-access-control-to-dataservices), they must authorize you to access the endpoints.
```
Comment (DivergentEuropeans) [inline docs/user-guide/mvd-configuration.md:678]: This link doesn't point to anything

**Review by lparadkar-rocket — APPROVED** [review]

**Pull request merged**